sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2002,6 +2007,11 @@ static String getExtractDestinationUri(String extractDestinationDir) {
       return toBuilder().setFormatFunction(formatFunction).build();
     }
 
+    /** Formats the user's type into a {@link TableRow} to be written to an error collector. */
+    public Write<T> withFailsafeFormatFunction(SerializableFunction<T, TableRow> formatFunction) {
```

Comment:
Just one change request: 
- Perhaps can we rename this to something that doesn't assume this fail-safe keyword is known by users? Perhaps `withFailedInsertFormatFunction`?
- And can you add some detail to the Javadoc so users will understand what this does?

Thanks

Reply:
Sounds good, renaming it to
`withFormatRecordOnFailureFunction`

And adding more in the Javadoc

Reply:
Done